docs/jobs.md

```diff
@@ -39,11 +39,21 @@ import socket
 def add(x, y):
     job = get_current_job()
     job.meta['handled_by'] = socket.gethostname()
-    job.save()
+    job.update_meta()
+    
+    # do more work
+    time.sleep(1)
     return x + y
 {% endhighlight %}
 
-
+<div class="warning">
+    <img style="float: right; margin-right: -60px; margin-top: -38px" src="{{site.baseurl}}img/warning.png" />
+    <strong>Warning!</strong>
+    <p>
+The `.update_meta()` was introduced in RQ 0.7.2. You can use the `.save()` method instead 
+for RQ <= 0.7.1. For that case you also have to set the job `ttl` greater than the job `timeout`, otherwise the job may dissappear from the queue during execution.
```

Comment:
In current implementation, `self.ttl == -1` during a job execution.  If I did the fix, then any `save()` call inside a job would change `ttl` value in the Redis key to -1.

I would not modify `job.ttl` inside a job at all and remove the implicit `cleanup` call from `save`. Of course, I would add explicit `cleanup` everywhere I need (inside worker.py, queue.py).

However, modification of the job stack may be useful also. `get_current_job` would be faster then.